### DISCUSSION TITLE
Why dark theme detector cannot detect dark theme on some websites?

### DISCUSSION THREAD

**Discussion by lumynou5** [darkreader/darkreader#12019] [Q&A] [answered]
I noticed that the implementation calculates lightnesses via the background colors of `body` and `html`, and some websites obviously have dark background colors; however the detector cannot detect them and it needs to be configure manually in `src/config/detector-hints.config`. I read #9016 and I guess the reason in this case is:
> 2. Some websites enable their own dark theme after Dark Reader's detection has already passed.

If it's correct, I wonder why. Because manually filling the list is inefficient and results a bad experience before the list is updated, is it possible to detect for a longer time to deal with this case?

**Comment by alexanderby** [comment, marked as answer]
In order to detect the dark theme we have to disable all Dark Reader's styles, and then enable them again. When there are a lot of styles loaded, it can cause performance issues, especially if it is run with some interval. Another approach will be published soon, and the detection will be improved periodically.
